components/automate-cli/pkg/verifyserver/models/populatewith_test.go

```diff
@@ -0,0 +1,432 @@
+	c = &Config{}
+	certsByIP = nil
+	ipList = []string{""}
+	privateKey = "private_key"
+	publicKey = "public_key"
+	adminKey = "admin_key"
+	adminCert = "admin_cert"
+
+	c.appendCertsByIpToNodeCerts(&certsByIP, ipList, privateKey, publicKey, adminKey, adminCert)
+
+	expected = []NodeCert{
+		{
+			IP:        "",
+			Key:       "private_key",
+			Cert:      "public_key",
+			AdminKey:  "admin_key",
+			AdminCert: "admin_cert",
+		},
+	}
+
+	// Check if each expected element exists in the actual result
+	for _, expectedNode := range expected {
+		found := false
+		for _, actualNode := range c.Certificate.Nodes {
+			if reflect.DeepEqual(expectedNode, actualNode) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("Test case 2 failed: Expected node %+v not found in the result.", expectedNode)
+		}
+	}
+}
+
+func TestPopulateWith(t *testing.T) {
+	tests := []struct {
+		name     string
+		filePath string
+		want     *Config
+		wantErr  bool
+		err      error
+	}{
+		{
+			name:     "PopulateWith Invalid OnPrem Config",
+			filePath: "./testdata/InvalidHaOnPrem.toml",
+			want:     &Config{},
+			wantErr:  true,
+			err:      errors.New("invalid or empty: ssh_user\ninvalid or empty: ssh_key_file\ninvalid S3 endpoint format. Endpoint should end with '.amazonaws.com'\nautomate private ip 1324.2534.1is not valid\ninvalid or empty: chef_server_private_ips\ninvalid or empty: opensearch_private_ips\ninvalid or empty: postgresql_private_ips\nurl should not include the protocol (http:// or https://): automate fqdn\nempty value: automate instance_count\ninvalid value 'automate.toml' for field 'config_file'. Expected values are: configs/automate.toml\ninvalid format. Failed to decode root_ca for automate\ninvalid format. Failed to decode private_key for automate\ninvalid format. Failed to decode public_key for automate\ninvalid format. Failed to decode private_key for automate ip\ninvalid format. Failed to decode public_key for automate ip\ninvalid value 'chef server instance_count' for field 'two'\ninvalid format. Failed to decode private_key for chef-server\ninvalid format. Failed to decode public_key for chef-server\ninvalid format. Failed to decode private_key for chef server ip\ninvalid format. Failed to decode public_key for chef server ip\nempty value: opensearch instance_count\nopensearch root_ca and/or admin_key and/or admin_cert and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false\nopensearch ippublic_key and/or private_key are missing in certs_by_ip. Otherwise set enable_custom_certs to false\nempty value: postgresql instance_count\ninvalid format. Failed to decode root_ca for postgresql\ninvalid format. Failed to decode private_key for postgresql\ninvalid format. Failed to decode public_key for postgresql\npostgresql ip 0.0.1 for certs is not valid\ninvalid format. Failed to decode private_key for postgresql ip\ninvalid format. Failed to decode public_key for postgresql ip"),
```

Comment:
First letter of error was capital before since it was showing warning, error strings should not be capitalized (ST1005) go-staticcheck, i made it lowercase.